src/sources.rs

```diff
@@ -116,6 +120,9 @@ impl Sources {
             let libs: BTreeMap<String, SourceConfig> = load_yaml(&path)?;
             for (lib_key, lib_info) in &libs {
                 let context = lib_info.context.value()?;
+                if *context != context.without_repository_subdirectory() {
+                    Err(ErrorKind::LibHasRepoSubdirectory(lib_key.clone()))?;
```

Comment:
I'm tempted to add a comment explaining something like, "We might actually be able to handle this case, but lib sources are already awkward enough without adding more features."

Reply:
👍